### PR TITLE
Cache `Image::getHtml()` to speed up the tree view

### PR DESCRIPTION
**Pull request by Toflar**
Huge tree views (or also list views) with lots of operations will slow down considerably at the moment because (amongst other stuff) expensive `Image::getHtml()` calls. I've tried to fix this using some kind of template per `$src` and then only replace the attributes and alt stuff dynamically.

Bildschirmfoto 2024-03-07 um 17 50 25

### CODE REVIEW

**Review by fritzmg — COMMENTED** [review]
Wouldn't Contao 4.13 benefit from this too? Since this is a legacy component it should be able to be merged upstream without any issues.

**Comment by leofeyer** [comment]
Which calls were the slowest? The `file_exists()` calls? Maybe we can add a file path cache for icons similar to the Twig template cache?

**Comment by Toflar** [comment]
The slowest are `getPath()` and `getSrc()` which in turn do call all sorts of other methods so I decided to cache the entire "template" and I'd suggest to keep this.
Contao 4.13 won't be affected as much because the whole dark mode handling does not exist.

**Comment by fritzmg** [comment]
> Contao 4.13 won't be affected as much because the whole dark mode handling does not exist.

Ah I see 👍

**Review by leofeyer — APPROVED** [review]

**Comment by leofeyer** [comment]
Thank you @Toflar.

File: core-bundle/contao/library/Contao/Image.php [hunk 1]
```diff
@@ -204,24 +240,9 @@ public static function getHtml($src, $alt='', $attributes='')
 				$darkSrc = substr($darkSrc, \strlen($webDir) + 1);
 			}
 
-			$darkAttributes = new HtmlAttributes($attributes);
-			$darkAttributes->mergeWith(array('class' => 'color-scheme--dark', 'loading' => 'lazy'));
-
-			foreach (array('data-icon', 'data-icon-disabled') as $icon)
```
Comment (zoglo) [inline core-bundle/contao/library/Contao/Image.php:210]: @Toflar you reintroduced #6869 in this one:
https://github.com/contao/contao/commit/3145d8b798c5191709cec250015b2ef9b1aa23e6

**Pull request merged**